config/config.go

```diff
@@ -849,6 +850,79 @@ func setupTranslation() error {
 	})
 }
 
+// findFieldByTag searches for a field in a struct by the value of a tag. This is used to
```

Comment:
Can you pull the two functions out from `config.go`? I'm worried that we put too much things in this single file. Since this is about config validation, probably put them in `validation.go`